includes/CLI/Plugin_Check_Command.php

```diff
@@ -687,4 +696,32 @@ function ( $item ) use ( $warning_severity ) {
 
 		return array_merge( $errors, $warnings );
 	}
+
+	/**
+	 * Returns check results filtered by severity level.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param array $results          Check results.
+	 * @param int   $error_severity   Error severity level.
+	 * @return array Filtered results.
+	 */
+	private function convert_filtered_errors_by_severity( $results, $error_severity ) {
```

Comment:
Implementation-wise, this could be much simplified and integrated into the `get_filtered_results_by_severity()` method. There we can do a simple `foreach` loop so we only loop through the messages once and avoid unnecessary `array_filter` and `array_merge` callse.